test-skripsi/test-improvement.tsx

```diff
@@ -23,11 +23,25 @@
 	}
 
 	getAllData(): Array<T> {
-		return this.data;
+		return [...this.data];
```

Comment:
Returning a shallow copy of the data array using `[...this.data]` is a great way to prevent accidental mutations to the internal state. Good job on ensuring data encapsulation.